test/test_arraycontext.py

```diff
@@ -672,36 +672,44 @@ def test_array_context_einsum_array_tripleprod(actx_factory, spec):
 # {{{ array container classes for test
 
 @with_container_arithmetic(bcast_obj_array=False,
-        eq_comparison=False, rel_comparison=False)
+        eq_comparison=False, rel_comparison=False,
+        _cls_has_array_context_attr=True)
 @dataclass_array_container
 @dataclass(frozen=True)
 class MyContainer:
     name: str
-    mass: DOFArray
+    mass: DOFArray   # or np.ndarray
```

Comment:
We need some work in `dataclass_array_container` to make this work with a `Union`, right?

Reply:
I seem to recall we tried, and it wasn't easy.